.github/workflows/bump-version.yml

```diff
@@ -1,44 +1,151 @@
+        run: |
+          if [[ "${{ github.event.workflow_run.conclusion }}" != "success" ]]; then
+            echo "### ❌ Previous Workflow Failed" >> $GITHUB_STEP_SUMMARY
+            echo "The triggering workflow did not complete successfully, please review the workflow run details." >> $GITHUB_STEP_SUMMARY
+            exit 1
+          else
+            echo "### ✅ Previous Workflow Succeeded" >> $GITHUB_STEP_SUMMARY
+          fi
+        shell: bash
+
+  check-token:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check GH_PUSH_TOKEN
+        run: |
+          if [[ -z "${{ secrets.GH_PUSH_TOKEN }}" ]]; then
+            echo "### ❌ GH_PUSH_TOKEN Error" >> $GITHUB_STEP_SUMMARY
+            echo "The \`GH_PUSH_TOKEN\` secret is not set, which is required for git operations." >> $GITHUB_STEP_SUMMARY
+            exit 1
+          else
+            echo "### ✅ GH_PUSH_TOKEN Found" >> $GITHUB_STEP_SUMMARY
+          fi
+        shell: bash
+
+  download-artifact:
+    needs: [check-previous-workflow, check-token]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Wait and download version artifact
+        run: |
+          attempt_number=1
+          max_attempts=10
+          artifact_name="version-${{ github.event.workflow_run.id }}"
+          while [ "$attempt_number" -le "$max_attempts" ]; do
+            echo "Attempt $attempt_number of $max_attempts"
+            response=$(curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+              -H "Accept: application/vnd.github.v3+json" \
+              "https://api.github.com/repos/${{ github.repository }}/actions/artifacts")
+            artifact_id=$(echo "$response" | jq --arg artifact_name "$artifact_name" '.artifacts[] | select(.name==$artifact_name) | .id')
+
+            if [ -n "$artifact_id" ]; then
+              echo "Artifact $artifact_name with ID $artifact_id is available"
+              # If available, download the artifact
+              curl -L -o "version-artifact.zip" \
+                -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+                "https://api.github.com/repos/${{ github.repository }}/actions/artifacts/$artifact_id/zip"
+              unzip -j "version-artifact.zip" "version.txt" -d .
+              # Check if the version.txt was extracted successfully
+              if [[ ! -f "version.txt" ]]; then
+                echo "### ❌ Failed to extract version.txt" >> $GITHUB_STEP_SUMMARY
+                exit 1
+              fi
+              break
+            else
+              echo "Artifact $artifact_name is not available yet"
+              if [ "$attempt_number" -eq "$max_attempts" ]; then
+                echo "### ❌ Artifact $artifact_name is not available after $max_attempts attempts" >> $GITHUB_STEP_SUMMARY
+                exit 1
+              fi
+              attempt_number=$((attempt_number+1))
+              echo "Waiting before retrying..."
+              sleep 10
+            fi
+          done
```

Comment:
Could this be simplified using the official [download artifact action](https://github.com/actions/download-artifact)? We're currently using it to cache the app build during e2e testing. I imagine this would be similar and might allow us to rely on the action for retry logic?

Reply:
As far as I can tell the action doesn't actually do any retry logic, but I need to double check.